### ISSUE TITLE
Query loop: Clarify the `Max page to show` setting

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#63028]
### Description

Noticed while working on https://github.com/WordPress/gutenberg/pull/63020

In the Query loop block, the 'Items per page', 'Offset', and 'Max page to show' settings are grouped together.
Currently, they're in a popover that opens from the block toolbar. In https://github.com/WordPress/gutenberg/pull/63020 they will be moved to the Inspector panel.

Regardlessof their overall placement, I'm not sure these settings should be grouped together.
- 'Items per page' and 'Offset' are more related to the Query properties.
- 'Max page to show' is more related to the Pagination links.

As a user, the 'Max page to show' setting isn't immediately clear to me. I'd tend to think this setting should be better placed, probably in its own section as suggested also by @carolinan 
Also, the description:

`Limit the pages you want to show, even if the query has more results. To show all pages use 0 (zero).`

doesn't clearly explain this setting is about _the pagination links_. While 'pagination' may be a technical term to avoid, I'd think this setting should be better explained.

Nnte: in https://github.com/WordPress/gutenberg/pull/63020 I'm fixing also a setting name typo where I guess 'page' should be plural 'pages' -> 'Max pages to show'.

### Step-by-step reproduction instructions

- Add a Query loop block > Start blank > Title & date.
- Click 'Display settings' in the block toolbar.
- Note: after https://github.com/WordPress/gutenberg/pull/63020 these settings will be placed in the Inspector panel.
- Observe the 'Max page to show' setting is grouped together with 'Items per page' and 'Offset'.
- Observe both the setting name and its description don't clarify very well this setting is related to the pagination links.

### Screenshots, screen recording, code snippet

![Screenshot 2024-07-01 at 15 41 32](https://github.com/WordPress/gutenberg/assets/1682452/6e83d7a5-ea38-4b62-8da8-6e81812b36a1)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Update: after https://github.com/WordPress/gutenberg/pull/58207 the 'Max pages to show' setting is now in the Inspector panel, hidden by default. Still, it's not clear it is about the amount of paginated pages.

![Screenshot 2024-09-20 at 10 41 02](https://github.com/user-attachments/assets/b7a4037c-7510-44e7-8378-2f2d07c61453)

**Comment by afercia** [comment]
See some additional feedback from https://github.com/WordPress/gutenberg/pull/65524#issuecomment-2374849973

> The "Max pages to show" setting was always confusing for me.

> * It has nothing to do with the query itself. It just sets the hard limit for the pages the pagination block can list.
> * It's not "reactive". When you change any other Query block setting, you get immediate feedback in the editor, but not for this one. Which makes it harder to connect a setting with visual output.

**Comment by ntsekouras** [comment]
It's true that this setting is used only `pagination numbers and next` blocks and after a bit of searching it seems it was added in the initial creation of Query Loop block. That was a way more simplified version of what we have now and as it seems it stuck there..

We could move this setting to Pagination block but I'm afraid that it's been to long there and might confuse users where to find it, if they need to.

So, do you think a dedicated inspector panel would help here? If yes, let's make this actionable and find the position and copy of the panel and message. --cc @jameskoster

**Comment by jameskoster** [comment]
I don't have a strong opinion about this one. Agree that conceptually it feels more like an option to include on the pagination block, but I expect some users would expect to find it on the Query Loop too. 

One thing we might consider is; removing the ability to toggle the option in ToolsPanel if the Query doesn't include a pagination block. Also; add the option to the pagination block and 'sync' it with the one on the Query, so you can edit it in both places? Like I said, I don't feel strongly about this, and I'm not sure it's worth the effort. In general the Query block could use a holistic design audit, like you mentioned many parts of it haven't been touched in years.

**Comment by afercia** [comment]
> removing the ability to toggle the option in ToolsPanel if the Query doesn't include a pagination bloc

I'd support this as a step to make the Query block usage clearer. It makes sense to not show settings that don't do anything.

> Also; add the option to the pagination block and 'sync' it with the one on the Query, so you can edit it in both places?

It could be a first step.

**Comment by afercia** [comment]
>  It makes sense to not show settings that don't do anything.

In a way, this point is also related to the fact this setting doesn't preview in the editor. Changing the setting doesn't show any change, which contributes to the confusion of what this setting does. See https://github.com/WordPress/gutenberg/issues/63027
